### DISCUSSION TITLE
Test install with only http rewrites URLs as https

### DISCUSSION THREAD

**Discussion by thepurpleblob** [directus/directus#10931] [Archive]
I am running a test/dev site on a local server using docker. I can access using localhost but attempting to access over the network fails with CORS and/or network errors. 

What seems to be happening is that I request address http://192.168.1.32:8055/admin.  (note, http)

However, the page fails to load with errors like The Same Origin Policy disallows reading the remote resource at https://192.168.1.32:8055/admin/assets/index.22c0e154.entry.js.  (note, now https)

I am unclear if where the https came from or (if that's nothing to do with it) how to sort the CORS issue. 

Thanks.

**Comment by thepurpleblob** [comment]
Bit more research.  The problem seems to be the header 'Upgrade-Insecure-Requests' which is set to '1'. So anything that doesn't specify the protocol is getting rewritten to https. Which is bad. 

Still unsure what to do about it.

**Comment by thepurpleblob** [comment]
So.... it turns out...

https://github.com/directus/directus/issues/10806

Adding the following to the config settings in docker-compose.yml fixed it

CONTENT_SECURITY_POLICY_DIRECTIVES__UPGRADE_INSECURE_REQUESTS: 'null'

Completely undocumented config option. Grrr

**Reply by rijkvanzanten** [reply]
> Completely undocumented config option. Grrr

https://docs.directus.io/configuration/config-options/#security for `CONTENT_SECURITY_POLICY` and https://docs.directus.io/configuration/config-options/#cors for `CORS`

**Comment by thepurpleblob** [comment]
We discussed some of this in the ticket I added. Although I note that the double underscore format *is* documented, I don't think you'd find it unless you already knew it was there. Unless of course, you read the documentation from end to end. Which is, of course, unthinkable ;-)

**Reply by rijkvanzanten** [reply]
Yup! And I noted that the double underscore isn't required there at all 😁
